notify/notify_test.go

```diff
@@ -91,44 +86,44 @@ func TestDedupStageNeedsUpdate(t *testing.T) {
 	now := utcNow()
 
 	cases := []struct {
-		entry    *nflogpb.Entry
-		hash     []byte
-		resolved bool
-		repeat   time.Duration
+		entry        *nflogpb.Entry
+		firingAlerts map[uint64]struct{}
+		repeat       time.Duration
 
 		res    bool
 		resErr bool
 	}{
 		{
-			entry: nil,
-			res:   true,
+			entry:        nil,
+			firingAlerts: map[uint64]struct{}{2: struct{}{}, 3: struct{}{}, 4: struct{}{}},
+			res:          true,
 		}, {
-			entry: &nflogpb.Entry{GroupHash: []byte{1, 2, 3}},
-			hash:  []byte{2, 3, 4},
-			res:   true,
+			entry:        &nflogpb.Entry{FiringAlerts: []uint64{1, 2, 3}},
+			firingAlerts: map[uint64]struct{}{2: struct{}{}, 3: struct{}{}, 4: struct{}{}},
```

Comment:
Add a small constructor function (inlined in the test function or elsewhere) to construct those sets? Probably cannot hurt overall to make a `alertHashSet` helper type hiding the awkwardness of `map[uint64]struct{}`.